sympy/solvers/ode/riccati.py

```diff
@@ -663,9 +663,9 @@ def compute_m_ybar(x, poles, choice, N):
 
     # Calculate the first (nested) summation for ybar
     # as given in Step 9 of the Thesis (Pg 82)
-    for i in range(len(poles)):
-        for j in range(len(choice[i])):
-            ybar += choice[i][j]/(x - poles[i])**(j+1)
+    for i, pole in enumerate(poles):
+        for j, c in enumerate(choice[i]):
+            ybar += c/(x - pole)**(j+1)
```

Comment:
If these are Expr then it is better to collect terms and sum the with `Add(*args)`. Iteratively adding in a loop is `O(n^2)` with Expr.